### PR TITLE
feat: Remove threshold from StatErrors

### LINKED ISSUE #31: Diverging behavior between StatErrors and CMS-treatment
Hey there :wave:

While digging deeper into the nll building blocks, I noticed that the behavior of `evm.staterror.StatErrors` probably needs a change to reach full parity with CMS' `autoMCStats` behavior, specifically the `include_signal` switch.

According to the [`autoMCStats` docs](https://cms-analysis.github.io/HiggsAnalysis-CombinedLimit/latest/part2/bin-wise-stats/#usage-instructions)

>  Note that this option only affects the logic for creating a single Barlow-Beeston-lite parameter vs. separate per-process parameters - the uncertainties on all signal processes are always included in the actual model!

So in any case, there should be MC stats. constraints on signal, they just do not enter the BB vs. BB-lite switch. If I see correctly, this is currently not possible with `evm.staterror.StatErrors`.

---

I think there would be two ways to solve this.

Either `StatErrors` receives two additional PyTree's for `hists` and `histsw2` to separate signal and background histograms, or it receives an additional filter (in whatever format that works best) to select only background contributions from `hists` and `histsw2`.

In any case, I think only the mask that decides between the two treatments would need to see an update:

https://github.com/pfackeldey/evermore/blob/fe47924cfccd1c93a36b9babd6c56a1e2d4d98a5/src/evermore/staterror.py#L75-L80

If you have a preference for any of the two (or a third way), I'm happy to prepare a quick PR.

### PR DESCRIPTION
**Pull request by riga**
Hi :wave:

As discussed in https://github.com/pfackeldey/evermore/issues/31, removing the BB/BB-lite threshold from `evm.staterror.StatErrors` would streamline the implementation and remove a statistical approximation used by CMS combine that is - strictly speaking - not necessary.

I removed the parameter and all related objects from `StatErrors` itself, as well as updated the docs accordingly. However, I kept the switch between "gaussian-per-process" and "poisson-per-process" as it is not altered by the threshold. I think it's debatable if it should be kept or not, but apparently it's also in use to avoid some edge cases.

Feel free up update wording and indentation!

Closes #31.

### CODE REVIEW

File: src/evermore/staterror.py [hunk 1]
```diff
@@ -55,31 +54,15 @@ class StatErrors(eqx.Module, SupportsTreescope):
...
-        self.threshold = threshold
-
-        self.ntot = sum_over_leaves(self.hists)
-        self.etot = jnp.sqrt(sum_over_leaves(self.histsw2))
-        ntot_eff = self.ntot**2 / self.etot**2
-        self.mask = ntot_eff > self.threshold
 
         # setup params
-        self.gaussians_global = NormalParameter(value=jnp.zeros_like(self.ntot))
         self.gaussians_per_process = jax.tree.map(
```
Comment (pfackeldey) [inline src/evermore/staterror.py:66]: I think we can even remove the gaussians per process, no? Then we only have poissons per process, or is there a good reason to keep those?

**Review by pfackeldey — APPROVED** [review]

**Comment by pfackeldey** [comment]
awesome thanks 🎉
